designcompose/src/main/java/com/android/designcompose/DesignView.kt

```diff
@@ -321,6 +324,10 @@ internal data class MaskInfo(
     val type: MaskViewType?,
 )
 
+// Custom Compose Semantic that identifies a DesignCompose Composable
+val docIdSemanticsKey = SemanticsPropertyKey<String>("DocId")
+var SemanticsPropertyReceiver.sDocId by docIdSemanticsKey
```

Comment:
I spent far too long reviewing these two lines because I wasn't familiar with the `var x by y` syntax, but I think this is just establishing `sDocId` as a global variable (so it's actually not very complicated, just very compact).

Reply:
Yeah, it's kind of wild. I'm not sure I fully grasp what happens there, it's code from https://developer.android.com/jetpack/compose/testing#custom-semantics-properties